content/lessons/chapter-3/coop-2.tsx

```diff
@@ -236,15 +236,14 @@ export default function Coop2({ lang }) {
                   {step === 0
                     ? 0
                     : (
-                        profile.nonce! /
-                        10 ** (profile.nonce!.toString().length - 2)
+                        profile.hashes! /
+                        10 ** (profile.hashes!.toString().length - 2)
```

Comment:
Rather than telling the typescript compiler were absolutely certain `profile.hashes` exists, I think it makes sense to add a type or interface instead that represents the protagonists/antagonists. 

Something like:
```ts
interface Player {
      username: string;
      avatar: string;
      hashpower: number;
      hashes: number;
      color: string;
      value: number;
}
```